src/allotropy/parsers/benchling_empower/benchling_empower_structure.py

```diff
@@ -79,94 +164,548 @@ def _get_chromatogram(injection: dict[str, Any]) -> DataCube | None:
+        )
+
+    return processing_items
+
+
+def _create_measurements(
+    injection: JsonData,
+    metadata_fields: JsonData,
+) -> list[Measurement]:
+    peaks = injection.data.get("peaks", [])
+    results = injection.data.get("results", [])
+
+    sample_type = injection.get(str, "SampleType")
+    sample_role_type = _get_sample_role_type(sample_type)
+    measurement_time = metadata_fields.get(str, "SampleSetStartDate")
+
+    measurement_custom_info = _extract_measurement_custom_info(
+        injection, metadata_fields
+    )
+    device_control_docs = _create_device_control_docs(injection)
+
+    processing_method_ids = set()
+    if results:
+        for result_item in results:
+            if method_id := result_item.get("ProcessingMethodId"):
+                processing_method_ids.add(method_id)
+
+    processing_methods = metadata_fields.data.get("processing_methods", [])
+    data_processing_by_method_id = (
+        _build_data_processing_by_method_id(processing_methods, processing_method_ids)
+        if processing_methods and processing_method_ids
+        else defaultdict(list)
+    )
+
+    processing_items = _build_processing_items(results, data_processing_by_method_id)
+
+    # Extract more custom info dictionaries
+    sample_custom_info = _extract_sample_custom_info(injection)
+    injection_custom_info = _extract_injection_custom_info(injection)
+
+    # Get required values
+    sample_name = injection[str, "SampleName"]
+    injection_id = injection[str, "InjectionId"]
+    date_acquired = injection[str, "DateAcquired"]
+
+    vial_id = injection.get(str, "VialId")
+    vial_id_str = str(vial_id) if vial_id is not None else None
+
+    injection.mark_read({"chrom", "peaks", "results", "curves"})
+
+    # Create and return the measurement
     return [
         Measurement(
             measurement_identifier=random_uuid_str(),
-            sample_identifier=assert_not_none(
-                injection.get("SampleName"), "SampleName"
-            ),
-            batch_identifier=injection.get("SampleSetID"),
+            sample_identifier=sample_name,
+            batch_identifier=str(metadata_fields.get(str, "sample_set_id") or ""),
             sample_role_type=sample_role_type,
-            written_name=injection.get("Label"),
-            chromatography_serial_num=injection.get("ColumnSerialNumber")
+            written_name=injection.get(str, "Label"),
+            chromatography_serial_num=injection.get(str, "ColumnSerialNumber")
             or NOT_APPLICABLE,
-            autosampler_injection_volume_setting=injection["InjectionVolume"],
-            injection_identifier=str(injection["InjectionId"]),
-            injection_time=injection["DateAcquired"],
-            peaks=[peak for peak in [_create_peak(peak) for peak in peaks] if peak],
+            autosampler_injection_volume_setting=injection.get(
+                float, "InjectionVolume"
+            ),
+            injection_identifier=str(injection_id),
```

Comment:
No need to cast to a string here. I also don't see it used elsewhere, so:
```suggestion
            injection_identifier=injection[str, "InjectionId"],
```
generally, we only get the value outside of the instantiation if we need to use it multiple times. Same goes for other required values in this case I think.